김진/231214/신규-아이디-추천.js

```diff
@@ -0,0 +1,18 @@
+function solution(new_id) {
+  new_id = new_id
+    .toLowerCase(/[A-Z]/g)
+    .replaceAll(/[^a-z0-9-_.]/g, "")
+    .replaceAll(/\.+/g, ".")
+    .replaceAll(/^\./g, "")
+    .replaceAll(/\.$/g, "");
+  new_id.length === 0
+    ? (new_id = "a")
+    : new_id.length >= 16
+    ? (new_id = new_id.substr(0, 15))
+    : new_id;
+  new_id = new_id.replaceAll(/\.$/g, "");
```

Comment:
```suggestion
    ? (new_id = new_id.substr(0, 15).replaceAll(/\.$/g, "");)
    : new_id;
```
이렇게 하면 코드가 간결해지고 한번에 처리가 가능할 것 같아요